cmd/kubeadm/app/cmd/phases/init/waitcontrolplane.go

```diff
@@ -119,3 +138,83 @@ func newControlPlaneWaiter(dryRun bool, timeout time.Duration, client clientset.
 
 	return apiclient.NewKubeWaiter(client, timeout, out), nil
 }
+
+type startupResult bool
+
+type startupProber struct {
+	runners []func(ch chan<- startupResult)
+}
+
+// getControlPlaneComponentStartupProber get all control plane component startup probes
+func getControlPlaneComponentStartupProber(cfg *kubeadm.InitConfiguration) startupProber {
+	specs := controlplane.GetStaticPodSpecs(&cfg.ClusterConfiguration, &cfg.LocalAPIEndpoint, nil)
+	prober := startupProber{}
+	for name, podSpec := range specs {
+		for _, container := range podSpec.Spec.Containers {
+			p := container.StartupProbe
```

Comment:
i think if the user has removed the probe with --patches, this code should not continue and just return success?

Reply:
Also user may change the startup prober type by --patches.